rust/sdk-processor/src/utils/starting_version.rs

```diff
@@ -42,6 +42,77 @@ pub async fn get_starting_version(
     ))
 }
 
+pub async fn get_minimum_last_success_version(
+    indexer_processor_config: &IndexerProcessorConfig,
+    conn_pool: ArcDbPool,
+    processor_names: Vec<String>,
+) -> Result<u64> {
+    let min_processed_version = get_processed_version_from_db(conn_pool.clone(), processor_names)
+        .await
+        .context("Failed to get minimum last success version from DB")?;
+
+    // If nothing checkpointed, return the `starting_version` from the config, or 0 if not set.
+    Ok(min_processed_version.unwrap_or(
+        indexer_processor_config
+            .transaction_stream_config
+            .starting_version
+            .unwrap_or(0),
+    ))
+}
+
+async fn get_processed_version_from_db(
```

Comment:
`get_processed_version_from_db` -> `get_min_processed_version_from_db` or move the min to the calling function